hal/halx86/generic/buildtype.c

```diff
@@ -12,3 +12,9 @@
 /* GLOBALS ******************************************************************/
 
 const USHORT HalpBuildType = HAL_BUILD_TYPE;
+
+#ifdef CONFIG_SMP
+KIRQL HalpIrqlSynchLevel = (IPI_LEVEL - 2);
+#else
+KIRQL HalpIrqlSynchLevel = DISPATCH_LEVEL;
+#endif
```

Comment:
ah, these "hacks" wouldn't have been necessary _**IF**_ the _**whole**_ HAL code was consistently recompiled with new definitions. Instead of trying to be too clever and compiling once parts of HAL code that happen to subtly depend on UP versus MP definitions.

Reply:
Yeah i know, I think eventually we should change it up a bit to not build libraries that make up the HALs. There's to many places where definitions don't line up.

But that's not for today i guess.